File: 프로그래머스/0/120861. 캐릭터의 좌표/캐릭터의 좌표.java
class Solution {
    public int[] solution(String[] keyinput, int[] board) {
        int[] answer = new int[2];
        int num = 0;
        int num2 = 0;
        for(int i = 0; i < keyinput.length; i++) {
            if(keyinput[i].equals("left")) {
                if(num == board[0] / 2 * -1) {
                    num = board[0] / 2 * -1;
                }
                else {
                    num -= 1;    
                }  
            }
            
            else if(keyinput[i].equals("right")) {
                if(num == board[0] / 2) {
                    num = board[0] / 2;
                }
                else {
                    num += 1;    
                }  
            }
            
            else if(keyinput[i].equals("up")) {
                if(num2 == board[1] / 2) {
                    num2 = board[1] / 2;
                }
                else {
                    num2 += 1;    
                }  
            }
            
            else if(keyinput[i].equals("down")) {
                if(num2 == board[1] / 2 * -1) {
                    num2 = board[1] / 2 * -1;
                }
                else {
                    num2 -= 1;    
                }  
            }
        }
        
        answer[0] = num;
        answer[1] = num2;
        return answer;
    }
}

File: 프로그래머스/0/120861. 캐릭터의 좌표/캐릭터의 좌표.js
function solution(keyinput, board) {
    // 초기 위치
    var x = 0; 
    var y = 0;

    // x와 y의 경계
    const maxX = Math.floor(board[0] / 2);
    const maxY = Math.floor(board[1] / 2);

    for (let i of keyinput) {       // 위로 이동
        if (i === 'up') {
            if (y < maxY) {
                y += 1; 
            }
        } else if (i === 'down') {      // 아래로 이동
            if (y > -maxY) {
                y -= 1; 
            }
        } else if (i === 'left') {      // 왼쪽으로 이동
            if (x > -maxX) {
                x -= 1; 
            }
        } else if (i === 'right') {     // 오른쪽으로 이동
            if (x < maxX) {
                x += 1; 
            }
        }
    }

    return [x, y];           // 최종 좌표 반환
}